### PR TITLE
[upgrade] eXist changes for next XML:DB API version

### PR DESCRIPTION
**Pull request by reinhapa**
This PR contains all changes done for the next [XMLDB:API](https://github.com/xmldb-org/xmldb-api) version.

### CODE REVIEW

File: exist-ant/src/main/java/org/exist/ant/RestoreTask.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
     private Path   dir             = null;
     private DirSet dirSet          = null;
     private String restorePassword = null;
-    private boolean overwriteApps = false;
+    private final boolean overwriteApps = false;
```
Comment (dizzzz) [inline exist-ant/src/main/java/org/exist/ant/RestoreTask.java:50]: interesting

File: exist-ant/src/main/java/org/exist/ant/XMLDBExtractTask.java [hunk 2]
```diff
@@ -133,8 +134,8 @@ public void execute() throws BuildException {
      * @throws IOException if an I/O error occurs
      */
     private void extractResources(final Collection base, final String path) throws XMLDBException, IOException {
-        final String[] resources = base.listResources();
-        if (resources != null) {
+        final List<String> resources = base.listResources();
+        if (!resources.isEmpty()) {
```
Comment (dizzzz) [inline exist-ant/src/main/java/org/exist/ant/XMLDBExtractTask.java:138]: no risk for nPE?

File: exist-core/src/main/java/org/exist/client/InteractiveClient.java [hunk 3]
```diff
@@ -377,39 +377,38 @@ protected void getResources() throws XMLDBException {
...
-        resources = new String[childCollections.length + childResources.length];
+        resources = new String[childCollections.size() + childResources.size()];
         //Collection child;
         Permission perm;
 
         final List<ResourceDescriptor> tableData = new ArrayList<>(resources.length); // A list of ResourceDescriptor for the GUI
 
         int i = 0;
-        for (; i < childCollections.length; i++) {
+        for (; i < childCollections.size(); i++) {
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/client/InteractiveClient.java:391]: looks like a better orlop can babe used, not using the index?

File: exist-core/src/main/java/org/exist/client/NewResourceDialog.java [hunk 4]
```diff
@@ -287,7 +287,7 @@ private void createResource(final ResourceType resourceType, final String filena
         
         try {
             final String resName = URIUtils.urlEncodeUtf8((isNullOrEmpty(filename) ? DEFAULT_FILENAME : filename) + "." + resourceType.getFileExtension());
-            final String resType = resourceType == ResourceType.XML_DOCUMENT ? XMLResource.RESOURCE_TYPE : BinaryResource.RESOURCE_TYPE;
+            final Class<? extends Resource> resType = resourceType == ResourceType.XML_DOCUMENT ? XMLResource.class : BinaryResource.class;
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/client/NewResourceDialog.java:290]: for readability split over lines?

**Review by dizzzz — COMMENTED** [review]
nice work

File: exist-core/src/main/java/org/exist/client/InteractiveClient.java (hunk 3 above)
Comment (reinhapa) [reply exist-core/src/main/java/org/exist/client/InteractiveClient.java:391]: I was too tiered to tackle this in the first pass  😊

File: exist-ant/src/main/java/org/exist/ant/RestoreTask.java (hunk 1 above)
Comment (reinhapa) [reply exist-ant/src/main/java/org/exist/ant/RestoreTask.java:50]: jep, guess this change was made by the IDE though. Still interesting though...

File: exist-ant/src/main/java/org/exist/ant/XMLDBExtractTask.java (hunk 2 above)
Comment (reinhapa) [reply exist-ant/src/main/java/org/exist/ant/XMLDBExtractTask.java:138]: The API states that there no null results are allowed instead an empty list must be returned.

File: exist-core/src/main/java/org/exist/client/NewResourceDialog.java (hunk 4 above)
Comment (reinhapa) [reply exist-core/src/main/java/org/exist/client/NewResourceDialog.java:290]: I don't seem to get your point here... The `resType` is now of type `Class` type not a `String` any more and the generic definition is needed in order to work for the later call.

Comment (line-o) [reply exist-core/src/main/java/org/exist/client/NewResourceDialog.java:290]: I believe @dizzzz was only referring to the formatting.
Maybe he had something like this in mind.

```
final Class<? extends Resource> resType = resourceType == ResourceType.XML_DOCUMENT 
    ? XMLResource.class
    : BinaryResource.class;
```

Comment (reinhapa) [reply exist-core/src/main/java/org/exist/client/NewResourceDialog.java:290]: I see... check for that with the next batch of changes then

**Comment by line-o** [comment]
@reinhapa do you plan to change the format of your commit messages to `[enhancement/bugifix/...] message` ?
More info on "Commit tagging" can be found in the [eXist-db Developer Manifesto]http://www.exist-db.org/exist/apps/doc/devguide_manifesto.xml

**Comment by reinhapa** [comment]
> @reinhapa do you plan to change the format of your commit messages to `[enhancement/bugfix/...] message` ? More info on "Commit tagging" can be found in the [eXist-db Developer Manifesto](http://www.exist-db.org/exist/apps/doc/devguide_manifesto.xml)

I have this in mind, the question is on how many commits I should reduce the existing ones, as there are some intermittent changes that will no longer compile in that way as the XML:DB API has changed over time. Feedback also from @adamretter  would be appreciated there.

**Pull request merged**